scripts/hardhat-fixture-deployer.ts

```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: 2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+const { resetNetworkToInitialState } = require('./common-test-snapshot');
+const deploy = require('../deploy/0_deploy');
+
+// Anonymous functions cannot be used as fixtures, hence we need to wrap body
+// in a method which will be called by `loadFixture`.
+async function deployAllContracts() {
+    await resetNetworkToInitialState();
+    await deploy();
+}
+
+export const loadHardhatFixtureDeployment = async () => {
+    console.log('Running hardhat-fixture');
+    await loadFixture(deployAllContracts);
+};
```

Comment:
It works for the first test but not for the following ones:
```
     FixtureSnapshotError: There was an error reverting the snapshot of the fixture.

This might be caused by using nested loadFixture calls in a test, for example by using multiple beforeEach calls. This is not supported yet.
```
:slightly_frowning_face: 
From what I understand (?) for the second test (as an example) if the network is reset outside (and before) the fixture is run, then the state related to the snapshotId to restore is probably lost and the fixture cannot load. Does it makes sense?

Reply:
I see yes.
In that case maybe we can rename `deployAllContracts()` to `resetNetworkAndDeployAllContracts()`.

Reply:
Good idea let me update that :+1:

Reply:
Here is [Rename to resetNetworkAndDeployAllContracts](https://github.com/iExecBlockchainComputing/PoCo/pull/59/commits/f50587e214044619995ae0b0c9f1995327c2b0f1).